### PR TITLE
[WFCORE-3752] Upgrading to MSC 1.4.2.Final - brings support for multi-value services

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.jboss.org/browse/WFCORE-3752
https://issues.jboss.org/browse/WFCORE-3763
https://github.com/wildfly/wildfly-proposals/pull/49

Brings in support for simplified multi-value services API.

### CODE REVIEW

**Comment by ropalka** [comment]
Retest this please

File: component-matrix/pom.xml [hunk 1]
```diff
@@ -92,7 +92,7 @@
         <version.org.jboss.logmanager.log4j-jboss-logmanager>1.1.4.Final</version.org.jboss.logmanager.log4j-jboss-logmanager>
         <version.org.jboss.marshalling.jboss-marshalling>2.0.5.Final</version.org.jboss.marshalling.jboss-marshalling>
         <version.org.jboss.modules.jboss-modules>1.8.1.Final</version.org.jboss.modules.jboss-modules>
-        <version.org.jboss.msc.jboss-msc>1.3.2.Final</version.org.jboss.msc.jboss-msc>
+        <version.org.jboss.msc.jboss-msc>1.4.1.Final</version.org.jboss.msc.jboss-msc>
```
Comment (jmesnil) [inline component-matrix/pom.xml]: Could you amend you commit and reference https://issues.jboss.org/browse/WFCORE-3763 please?
JBoss MSC is a key component of WildFly and we should track its upgrade in separate WFCORE issue.

**Comment by jmesnil** [comment]
@ropalka The PR loos good to me. I have added the missing-reqs flag to keep this PR to be merged until it fulfills all missing requirements:

- [ ] approval of the https://github.com/wildfly/wildfly-proposals/pull/49 proposal
- [ ] community docs
- [ ] test plan

File: component-matrix/pom.xml (hunk 1 above)
Comment (ropalka) [reply component-matrix/pom.xml]: Done @jmesnil - thanks for pointing it out.

**Comment by jmesnil** [comment]
@ropalka is there some documentation for JBoss MSC (outside of WildFly)? The only one I found https://docs.jboss.org/author/display/MSC/Home is empty.

I'm not sure which kind of documentation we are expecting for such a developer feature.
The feature is explained in ServiceBuillder Javadoc at https://github.com/jboss-msc/jboss-msc/blob/13fb3be15177db8f812903f8d859ee2ad96112df/src/main/java/org/jboss/msc/service/ServiceBuilder.java#L33 so the developers writing Services will know how to leverage the feature.

That seems fine to me.

@kabir What do you think?

**Comment by ropalka** [comment]
https://github.com/wildfly/wildfly-proposals/pull/49 have been approved @jmesnil

**Comment by bstansberry** [comment]
@ropalka I agree with @jmesnil that javadoc is ok for this as far as the community doc requirement goes. If there already was an MSC guide I wouldn't feel that way but I don't think it's reasonable to require such a guide to be created just to get this change in.

So, we need "test plan" and "QE preverification".  Please get @simkam to approve this and that's covered.

**Comment by kabir** [comment]
Sorry @jmesnil I didn't see your ping, I agree we should not block this on creating a new guide (especially since one did not exist)

**Comment by jmesnil** [comment]
@ropalka since the conflict was 2 lines in the component-matrix/pom.xml, I tried to use GitHub UI to solve this (it was the first time I tried it) but it adds a new commit for it. Could you squash the 2 commits in a single one please?

thanks!

**Comment by kabir** [comment]
@jmesnil Next to the 'Merge pull request' button there is a drop down you can use to do squash and merge. Also, I believe as a committer you can force push to ropalka's branch

**Comment by jmesnil** [comment]
the PR is ready to merge (I'll squash the 2 commits when I merge it)

**Pull request merged**